src/Logikfabrik.Umbraco.Jet/DataTypeSynchronizer.cs

```diff
@@ -124,21 +126,31 @@ private static DataTypeDatabaseType GetDatabaseType(Type type)
             return type == typeof(DateTime) ? DataTypeDatabaseType.Date : DataTypeDatabaseType.Ntext;
         }
 
+        private static IDictionary<string, PreValue> GetDataTypePrevaluesDictionary(DataType model)
+        {
+            if (model.PreValues == null || !model.PreValues.Any())
+            {
+                return new Dictionary<string, PreValue>();
+            }
+
+            return model.PreValues.ToDictionary(preValue => preValue.Key, v => new PreValue(v.Value));
+        }
+
         private void Synchronize(IDataTypeDefinition[] dataTypes, DataType model)
         {
             var dataType = _dataTypeFinder.Find(model, dataTypes).SingleOrDefault();
 
-            dataType = dataType == null
-                ? CreateDataType(model)
-                : UpdateDataTypeDefinition(dataType, model);
-
-            _dataTypeService.Save(dataType);
+            if (dataType == null)
+            {
+                // create new data type
+                _dataTypeService.SaveDataTypeAndPreValues(CreateDataType(model), GetDataTypePrevaluesDictionary(model));
+                return;
+            }
 
-            // We get the data type once more to refresh it after saving it.
-            dataType = _dataTypeService.GetDataTypeDefinitionByName(dataType.Name);
+            _dataTypeService.Save(UpdateDataTypeDefinition(dataType, model));
 
-            // Set the pre-values, if any.
-            SetDataTypePreValues(dataType, model);
+            var existingPreValues = _dataTypeService.GetPreValuesCollectionByDataTypeId(dataType.Id);
+            _dataTypeService.SavePreValues(dataType.Id, existingPreValues.AddOrUpdate(model.PreValues));
 
             // Set/update tracking.
             SetDataTypeId(model, dataType);
```

Comment:
If a new datatype is created, this method won't be called. If not called tracking will be broken until the datatype is updated.